service/dispatch/internal/broker/event_broker.go

```diff
@@ -44,11 +45,27 @@ func (b *eventBroker) CreateSubscription(sub service.Subscription) {
 		return
 	}
 
-	if err = b.broker.PublishAsync(CreateSubscriptionSubject, data); err != nil {
-		b.logger.Errorf(
-			"failed to publish CreateSubscription message to '%s' subject: %v",
-			CreateSubscriptionSubject, err,
-		)
+	subject := SubscriptionCreatedSubject
+	if err = b.broker.PublishAsync(subject, data); err != nil {
+		b.logger.Errorf("failed to publish message to '%s' subject: %v", subject, err)
+	}
+}
+
+func (b *eventBroker) CancelSubscription(sub service.Subscription) {
```

Comment:
Try to use just Publish method that accepts interface{} and use something like Reflection to get  event type. Also you can create domain events for this, so you can distinguish events from each other. For example 'SubscriptionCreated' 'SubscrtiptionCanceled' and you can use reflection on them. So for every new event, you won't create new methods.

Reply:
Rewrited eventBroker so it uses only single method `Publish(interface{})`

Reply:
Ok, i see your point. Good refactoring.